tests/unit/frameworks/pytorch/resnet_serialize_test.py

```diff
@@ -46,23 +46,27 @@ def test_identical(self):
             init_batch_norm=False,
             device=device,
         )
-        # create temp file
-        model_path = tempfile.NamedTemporaryFile(delete=True)
-        # and write the created model
-        torch.save(model.state_dict(), model_path)
 
-        model_copy = create_model(
-            model_class=model_class,
-            model_args=model_args,
-            init_batch_norm=False,
-            device=device,
-        )
-        # copy the initial model parameters to the new model
-        model_copy.load_state_dict(torch.load(model_path.name))
+        state = {}
+        with io.BytesIO() as buffer:
+            serialize_state_dict(buffer, model.state_dict())
+            state["model"] = buffer.getvalue()
+
+        with tempfile.NamedTemporaryFile(delete=True) as checkpoint_file:
+            pickle.dump(state, checkpoint_file)
+            checkpoint_file.flush()
+
+            model2 = create_model(
+                model_class=model_class,
+                model_args=model_args,
+                init_batch_norm=False,
+                device=device,
+                checkpoint_file=checkpoint_file.name
+            )
 
-        # comapre the initial and new model
-        self.assertTrue(compare_models(model, model_copy, input_shape=(3, 224, 224)))
-        model_path.close()
+            checkpoint_file.close()
```

Comment:
This is unnecessary. The checkpoint_file will be closed by the context manager. See https://docs.python.org/3/library/tempfile.html